ts/hooks/useValidateEmailModal.tsx

```diff
@@ -30,11 +32,20 @@ export const useValidatedEmailModal = (isOnboarding?: boolean) => {
 
   useFocusEffect(
     React.useCallback(() => {
-      if (!isEmailValidated) {
+      // AS-IS FLOW
+      if (!isNewCduFlow && !isEmailValidated) {
         showModal(<RemindEmailValidationOverlay isOnboarding={isOnboarding} />);
+        return () => hideModal();
+        // CDU FLOW
+      } else if (isNewCduFlow && !isEmailValidated) {
+        showModal(
+          <NewRemindEmailValidationOverlay isOnboarding={isOnboarding} />
+        );
+        return;
+        // OTHER CASES
+      } else {
+        return;
       }
-
-      return () => hideModal();
-    }, [hideModal, isOnboarding, showModal, isEmailValidated])
+    }, [hideModal, isEmailValidated, isOnboarding, showModal])
```

Comment:
```ts
  useFocusEffect(
    React.useCallback(() => {
      // AS-IS FLOW
      if (!isNewCduFlow && !isEmailValidated) {
        showModal(<RemindEmailValidationOverlay isOnboarding={isOnboarding} />);
        return () => hideModal();
        // CDU FLOW
      } else if (isNewCduFlow && !isEmailValidated) {
        showModal(
          <NewRemindEmailValidationOverlay isOnboarding={isOnboarding} />
        );
      }
      return () => void 0;
    }, [hideModal, isEmailValidated, isOnboarding, showModal])
  );
```

This, should do the same trick. Do you like it more?

Reply:
Yes, a lot! Thanks!